testcases/cloud_admin/modify_instance_type.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env python
+#
+# Description: Test the euca-modify-instance-type command works properly.
+#
+# NOTE: This testcase DOES NOT work with credentials file (credpath and eucarc) it only
+#       allows password/config_data for the time being.
+#
+from eucaops import Eucaops
+from eutester.eutestcase import EutesterTestCase
+
+import re
+import random
+
+class instance_type():  # mimicing a struct, I think this is the preferred way in python
+
+    # shaon pointed out that I should be using Vm_Type instead of my own class
+    # but can't figure that out right now - 4.28.2015 mbacchi
+    def __init__(self, name=None, cpus=None, mem=None, disk=None):
+        self.name = name
+        self.cpus = cpus
+        self.mem = mem
+        self.disk = disk
+
+    def p(self):
+        print ("Name: " + self.name + " cpu: " + str(self.cpus) + " mem: "
+               + str(self.mem) + " disk: " + str(self.disk))
+
+class InstanceTypeOps(EutesterTestCase):
+    _known_types = []
+
+    def __init__(self, name="InstanceTypeOps", credpath=None, region=None, config_file=None, password=None, emi=None, zone=None,
+                  user_data=None, instance_user=None, **kwargs):
+
+        super(InstanceTypeOps, self).__init__(name=name)
+        self.get_args()
+        #self.show_args()
+        for kw in kwargs:
+            print 'Setting kwarg:'+str(kw)+" to "+str(kwargs[kw])
+            self.set_arg(kw ,kwargs[kw])
+        #self.show_args()
+        self.setuptestcase()
+        self.setup_parser()
+        #self.get_args()
+        # Setup basic eutester object
+        self.tester = Eucaops(config_file=self.args.config_file,
+                                  password=self.args.password)
+                                  # credpath=self.args.credpath)  # <- credpath disabled since not functioning for these commands
+
+        self.PopulateInstanceTypes()
+        #print "Now printing _my_types after populating:"    # DEBUG
+        #for i in InstanceTypeOps._known_types:
+            #i.p()
+
+    def clean_method(self):
```

Comment:
We need to implement something here for cleanup if we have changed anything during the test cases run. Maybe we can store the state of the instance types before this test ran and then set them back at the end.